src/devices/Common/System/Device/DelayHelper.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Device
+{
+    /// <summary>
+    /// Helpers for short waits.
+    /// </summary>
+    internal static class DelayHelper
+    {
+        // GetTimestamp() currently can take ~300ns. We hope to improve this to get better
+        // fidelity for very tight spins.
+        //
+        // SpinWait currently spins to approximately 1μs before it will yield the thread.
+
+        /// <summary>
+        /// Delay for at least the specified <paramref name="microseconds"/>.
+        /// </summary>
+        /// <param name="microseconds">The number of microseconds to delay.</param>
+        /// <param name="allowThreadYield">True to allow yielding the thread. On single-proc systems this will prevent all other code from running.</param>
```

Comment:
I think the comment is was saying the opposite of what it should.   
```suggestion
        /// <param name="allowThreadYield">True to allow yielding the thread. If this is set to false, on a single-proc systems this will prevent all other code from running.</param>
```